llpc/translator/lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -49,11 +49,17 @@
 using namespace std;
 using namespace SPIRVDebug::Operand;
 
+namespace llvm {
+namespace cl {
+extern opt<bool> TrimDebugInfo;
```

Comment:
Use an external bool instead of `extern opt<bool>`: https://llvm.org/docs/CommandLine.html#internal-vs-external-storage
The llvm convention would be to call it `bool XTirimDebugInfo`.

Or even better, could we make this a constructor parameter to avoid global state altogether?

Reply:
>"Or even better, could we make this a constructor parameter to avoid global state altogether?"
m_dbgTran(m_bm, m_m, this) in constructed from SPIRVReader.  that might be a long chain of parameter passing from llpcCompiler where "TirimDebugInfo" is defined.

Reply:
Do you mean i need to add new definition

 bool XTirimDebugInfo, in the llpcCompiler.cpp

and modify opt<bool> TrimDebugInfo("trim-debug-info", cl::desc("Trim debug information in SPIR-V binary"), init(true));
with cl::location(XTirimDebugInfo)

then extern XTirimDebugInfo in the SPIRVToLLVMDbgTran.cpp

I figure it's more straighward to extern  opt<bool> TirimDebugInfo directly in the SPIRVToLLVMDbgTran.cpp without changing llpcCompiler.cpp

Reply:
> Do you mean i need to add new definition

> bool XTirimDebugInfo, in the llpcCompiler.cpp

Yes, this is what I had in mind

Reply:
maybe we can leave llpcCompiler alone and use TrimDebugInfo directly,  it's quite common practice in the Llpc code.

Reply:
OK, up to you. It's better to stay consistent.